packages/remix/src/executors/build/build.impl.ts

```diff
@@ -81,10 +82,14 @@ export default async function buildExecutor(
   }
 
   if (options.generateLockfile) {
-    const lockFile = createLockFile(packageJson);
-    writeFileSync(`${options.outputPath}/${getLockFileName()}`, lockFile, {
-      encoding: 'utf-8',
-    });
+    const lockFile = createLockFile(packageJson, context.projectGraph);
+    writeFileSync(
+      `${options.outputPath}/${getLockFileName(detectPackageManager(context.root))}`,
```

Comment:
The result from `detectPackageManager(context.root))` should go into `createLockFile` for consistency.

```js
const packageManager = detectPackageManager(context.root);
const lockFile = createLockFile(packageJson, context.projectGraph, packageManager);
    writeFileSync(
      `${options.outputPath}/${getLockFileName(packageManager)}`,
// ...
```